lib/xandra/protocol.ex

```diff
@@ -759,4 +771,15 @@ defmodule Xandra.Protocol do
     {elem, buffer} = decode_string(buffer)
     decode_string_list(buffer, size - 1, [elem | acc])
   end
+
+  defp decode_type_udt(buffer, 0, acc) do
+    {{:udt, acc}, buffer}
+  end
+
+  defp decode_type_udt(buffer, count, acc) do
+    {name, buffer} = decode_string(buffer)
+    {type, buffer} = decode_type(buffer)
+
+    decode_type_udt(buffer, count - 1, acc ++ [{name, type}])
```

Comment:
We almost never do `list ++ [elem]` in Elixir. The reasoning is explained in [the docs for `++/2`](https://hexdocs.pm/elixir/Kernel.html#%20%20/2): `a ++ b`'s complexity is proportional to `length(a)` (because of how linked lists work), so in your case, the more `acc` grows, the slower it will become to do `++`. The very very common pattern, mentioned in the docs I linked, is to do `[elem | list]` to prepend the element to the list (which is **constant-time**) and then reverse the list at the end of the recursion (reversing is implemented in C in Erlang, so it's fast-ish).

```elixir
defp decode_type_udt(buffer, 0, acc) do
  {{:udt, Enum.reverse(acc)}, buffer}
end

defp decode_type_udt(buffer, count, acc) do
  ...
  decode_type_udt(buffer, count - 1, [{name, type} | acc])
end
```